error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/IsInstanceLambdaUsageTest.java

```diff
@@ -23,6 +23,7 @@ void identification() {
             "    // BUG: Diagnostic contains:",
             "    Stream.of(1).filter(i -> i instanceof Integer);",
```

Comment:
The following expression is also flagged, while it shouldn't:
```java
Stream.of(1).filter(i -> someOtherLocalVariable instanceof Integer);
```

Similarly, we should not flag `BiPredicate`s such as in:
```java
Flux.just(1, "foo").distinctUntilChanged(v -> v, (a, b) -> a instanceof Integer);

Reply:
While there: in `identification` tests we often list the "good" cases first; I'll shuffle om things.